### PR TITLE
Allowed IP_FREEBIND option for UDP epoll

### PR DESCRIPTION
**Pull request by yulianoifa-mobius**
Motivation:

While using Load Balancers or HA support is needed there are cases when UDP channel need to bind to IP Address which is not available on network interfaces locally. 

Modification:

Modified EpollDatagramChannelConfig to allow IP_FREEBIND  option

Result:

Fixes ##8727.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@yulianoifa-mobius thanks a lot! Can you please also add a unit test that verifies you can set the option now ? Also please sign our ICLA: https://netty.io/s/icla

**Comment by normanmaurer** [comment]
Something like:

```java

public class EpollDatagramChannelConfigTest {

    @Test
    public void testIpFreeBind() throws Exception {
        Epoll.ensureAvailability();
        EpollDatagramChannel channel = new EpollDatagramChannel();
        assertTrue(channel.config().setOption(EpollChannelOption.IP_FREEBIND, true));
        assertTrue(channel.config().getOption(EpollChannelOption.IP_FREEBIND));

        channel.fd().close();
    }
}
```

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by yulianoifa-mobius** [comment]
@normanmaurer Hi , Sorry for late response i am currently traveling. I have signed the ICLA. As for test - i see that you have already added the tests.
Thanks and BR
Yulian Oifa

**Comment by normanmaurer** [comment]
@yulianoifa-mobius I did not add the test yet... just adjust add the test to the PR and I can merge :)

File: transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannelConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2015 The Netty Project
```
Comment (normanmaurer) [inline transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannelConfigTest.java]: @yulianoifa-mobius please change to 2019

File: transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannelConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    @Test
+    public void testIpFreeBind() throws Exception {
+        Epoll.ensureAvailability();
+        EpollDatagramChannel channel = new EpollDatagramChannel();
+        assertTrue(channel.config().setOption(EpollChannelOption.IP_FREEBIND, true));
+        assertTrue(channel.config().getOption(EpollChannelOption.IP_FREEBIND));
+
+        channel.fd().close();
+    }
+}
```
Comment (normanmaurer) [inline transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannelConfigTest.java]: Add a CRLF.

**Comment by normanmaurer** [comment]
@netty-bot test this please

File: transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannelConfigTest.java (hunk 2 above)
Comment (yulianoifa-mobius) [reply transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDatagramChannelConfigTest.java]: i guess you meaned to remove CRLF

**Comment by normanmaurer** [comment]
@netty-bot Test this please

**Comment by normanmaurer** [comment]
@yulianoifa-mobius please fix the checkstyle error:

23:08:51 /code/transport-native-epoll/src/test/java/io/netty/channel/epoll/EpollDatagramChannelConfigTest.java:0: File does not end with a newline.

**Comment by normanmaurer** [comment]
@netty-bot Test this please

**Comment by normanmaurer** [comment]
@yulianoifa-mobius thanks so much!

**Pull request merged**